Add an uninstall.sh

ialokim opened the pull request:
Up to now, there's an `install.sh` script that does the installation process, but it lacks uninstall functionality. I've created a little `uninstall.sh` to fill this gap.

Fixes #237

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+
+# Remove all fonts to user fonts directory
+echo "Removing fonts..."
+eval $find_command | xargs -n1 -0 -I % sh -c 'test="%"; rm -f "'"$font_dir"'/${test##*/}"'
```
I do not see why you should use `eval $find_command` here, but I can see why you should not: `"$var"` is not a proper escaping. Also don’t put directory names inside an argument via `%`, `sh -c` does accept arguments and this variant of passing arguments has no problems with escaping. Also don’t use `"$font_dir"` there as well, export variable instead.

ialokim replied:
The `eval` is necessary to support the `or` switch in the command string. But I've exported the variables as pointed out.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+eval $find_command | xargs -n1 -0 -I % sh -c 'test="%"; rm -f "'"$font_dir"'/${test##*/}"'
+
+# Reset font cache on Linux
+if command -v fc-cache @>/dev/null ; then
```
Why not `which` and what `@>` is supposed to mean? `which fc-cache >/dev/null 2>&1` normally works fine and if you spell that this way (and replace `[[ … ]]` with `test` above) then script will be free of bashism.

ialokim replied:
I'm okay with this.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Set source and target directories
+powerline_fonts_dir=$( cd "$( dirname "$0" )" && pwd )
```
Need quotes to be consistent. I know that they are not actually needed in variable assignments, but better have them then not.

ialokim replied:
Added.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+# Set source and target directories
+powerline_fonts_dir=$( cd "$( dirname "$0" )" && pwd )
+
+# if an argument is given it is used to select which fonts to install
```
Comment talks about installing.

ialokim replied:
Fixed.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+
+find_command="find \"$powerline_fonts_dir\" \( -name '$prefix*.[o,t]tf' -or -name '$prefix*.pcf.gz' \) -type f -print0 "
+
+if [[ `uname` == 'Darwin' ]]; then
```
Use `$()` to stay consistent with `$()` above.

ialokim replied:
Done.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+else
+  # Linux
+  font_dir="$HOME/.local/share/fonts"
+  mkdir -p $font_dir
```
You are doing uninstallation, if font directory does not exists *exit* with a message, not create it.

ialokim replied:
You're completely right, fixed that!

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,30 @@
+# Reset font cache on Linux
+if command -v fc-cache @>/dev/null ; then
+    echo "Resetting font cache, this may take a moment..."
+    fc-cache -f $font_dir
```
Missing quotes.

ialokim replied:
Fixed.

ZyX-I commented on README.rst:
```diff
@@ -25,6 +25,12 @@ You can copy and paste the commands to your terminal. Comments are fine too.
+Uninstall
+---------
+
+Run ``./uninstall.sh`` to uninstall all Powerline Fonts. You can also copy
```
Missing the fact that `./uninstall.sh` needs to be run with the same HEAD in the repository as fonts were installed, fonts are sometimes being removed from repository or renamed.

ialokim replied:
Fixed.

ialokim commented:
Thanks for your review, I've fixed your issues, while three of them were also present in the current `install.sh`. I've added a second commit fixing them, too.

ZyX-I commented:
> The eval is necessary to support the or switch in the command string. But I've exported the variables as pointed out.

How it is? It should work fine with double-quoted `$prefix` if you move replace eval with actual command. BTW, what does comma do in `.[o,t]tf`?

ialokim commented:
> How it is? It should work fine with double-quoted $prefix if you replace eval with actual command. BTW, what does comma do in .[o,t]tf?

As already stated, I've copied essentially the `install.sh` and changed only the necessary lines to provide uninstall functionality. I've now replaced the `eval` with the actual command as you've suggested and changed the `install.sh` accordingly, too.

ZyX-I commented on install.sh:
```diff
@@ -1,14 +1,12 @@
-find_command="find \"$powerline_fonts_dir\" \( -name '$prefix*.[o,t]tf' -or -name '$prefix*.pcf.gz' \) -type f -print0"
-
-if [[ `uname` == 'Darwin' ]]; then
+if test $(uname) == 'Darwin' ; then
```
`test` uses single equals sign. And `$()` needs to be escaped, or there is some small potential to get “missing argument after Darwin” error (`$(uname)` will expand into nothing if `uname` is missing).

ialokim replied:
Fixed.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,31 @@
+# if an argument is given it is used to select which fonts to uninstall
+prefix="$1"
+
+if test $(uname) == 'Darwin' ; then
```
Same here.

ZyX-I commented on uninstall.sh:
```diff
@@ -0,0 +1,31 @@
+else
+  # Linux
+  font_dir="$HOME/.local/share/fonts"
+  if test ! -d "$font_dir"; then
```
Above there is space before semicolon.

ialokim replied:
Added the space here, too.
